### ISSUE TITLE
Existential Types error doesn't point to offending location in project 

### ISSUE THREAD

**Issue by rbscgh** [lampepfl/dotty#19672]
## Compiler version 3.3.1

If you're not sure what version you're using, run `print scalaVersion` from sbt
(if you're running scalac manually, use `scalac -version` instead).

## Minimized example

The compiler is complaining about a usage of `forSome` however it is not found in either the project or its dependencies.

here are some cross compiled deps:
| Dep | Version |
|--------|--------|
| specs2 | 4.20.5 |
| specs2-mock | 4.20.5 |
| circe-core,generic,parser | 0.14.6 |
| monix | 3.4.1 |

## Output Error/Warning message

```scala
[warn] Explanation
[warn] ===========
[warn] Existential types in their full generality are no longer supported.
[warn] Scala-3 does applications of class types to wildcard type arguments.
[warn] Other forms of existential types that come from Scala-2 classfiles
[warn] are only approximated in a best-effort way.
[warn] An existential type that came from a Scala-2 classfile for trait ClassesOf
[warn] cannot be mapped accurately to a Scala-3 equivalent.
[warn] original type    : Seq[Class[? >: _$1 & _$1² & _$1³]] forSome _$1; _$1²; _$1³
[warn] reduces to       : Seq[Class[? >: _$1 & _$1² & _$1³]]
[warn] type used instead: Seq[Class[? >: Any]]
[warn] This choice can cause follow-on type errors or hide type errors.
[warn] Proceed at own risk.
[warn] 
[warn] where:    _$1  is a type in method classesOf
[warn]           _$1² is a type in method classesOf
[warn]           _$1³ is a type in method classesOf
[warn] 
[warn]   |-----------------------------------------------------------------------------
[warn]   | Explanation (enabled by `-explain`)
[warn]   |- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
[warn]   | Existential types in their full generality are no longer supported.
[warn]   | Scala-3 does applications of class types to wildcard type arguments.
[warn]   | Other forms of existential types that come from Scala-2 classfiles
[warn]   | are only approximated in a best-effort way.
[warn]    -----------------------------------------------------------------------------
```

## Why this Error/Warning was not helpful

No where in the direct project or dependency do I see a usage of forSome. It would be nice to see where the compiler is finding that issue so we can resolve it. 

## Suggested improvement

It could be made more helpful by including the offending line in project or dependency.

**Comment by smarter** [comment]
I believe the forSome here comes from the result type inferred by Scala 2 for the classesOf method and should disappear if an explicit result type of `Seq[Class[_]]` is used. Things Scala 3 could do better here:
- don't warn if the approximation only involves local variables like `_$1` ?
- Mention that this might be an inferred result type? I don't think the scala 2 pickled format gives us enough information to know this for sure.

**Comment by smarter** [comment]
> It could be made more helpful by including the offending line in project or dependency.

Agreed, but here again this might be tricky to reconstruct from the scala 2 pickled information.

**Comment by rbscgh** [comment]
@smarter because this is happening only when compiling tests.. do you know how i could suppress the error so i can maintain fatalWarnings for the rest of the code ?

**Comment by rbscgh** [comment]
https://github.com/scala/scala3/blob/7d559ad06bdfe448450341c7fdf12cc0513456f3/compiler/src/dotty/tools/dotc/reporting/messages.scala#L1924

here is the message in dotty, but how to find out the format for WConf-
